runtime/jvmti/jvmtiStartup.c

```diff
@@ -701,6 +707,13 @@ loadAgentLibrary(J9JavaVM * vm, J9JVMTIAgentLibrary * agentLibrary)
 
 	Trc_JVMTI_loadAgentLibrary_Entry(agentLibrary->nativeLib.name);
 
+	/* If dynamic agent loading is disabled, the HCR/OSR flags default to off. */
+	omrthread_monitor_enter(vm->runtimeFlagsMutex);
+	if (J9_ARE_NO_BITS_SET(vm->runtimeFlags, J9_RUNTIME_ALLOW_DYNAMIC_AGENT)) {
+		vm->extendedRuntimeFlags |= (J9_EXTENDED_RUNTIME_OSR_SAFE_POINT | (UDATA)J9_EXTENDED_RUNTIME_ENABLE_HCR);
```

Comment:
This doesn't do what the comment says: "flags default to off"; I think you meant
```c
		vm->extendedRuntimeFlags &= ~(UDATA)(J9_EXTENDED_RUNTIME_OSR_SAFE_POINT | J9_EXTENDED_RUNTIME_ENABLE_HCR);
```

Reply:
`omrthread_monitor_enter`/`omrthread_monitor_exit` pair can be moved into the block after `J9_ARE_NO_BITS_SET()` checking.

Reply:
> This doesn't do what the comment says: "flags default to off"

The code is correct, you are misunderstanding the comment. The comment means that the flags were off because J9_RUNTIME_ALLOW_DYNAMIC_AGENT wasn't set, so now they need to be turned on.

Reply:
That I misunderstood the comment occurred to me later. I think it can be improved.

But this just begs different questions.
* `J9_EXTENDED_RUNTIME_OSR_SAFE_POINT`, for example is set by default in `processVMArgsFromFirstToLast()`, but cleared if `-XX:-OSRSafePoint` or `-XX:-OSRSafePointFV` is specified; why should that flag be set here?
* Same question for `J9_EXTENDED_RUNTIME_ENABLE_HCR`.
* Why should we be here at all if dynamic agent loading is disabled?

Reply:
The quick answer is that this is what @tajila asked for in https://github.com/eclipse-openj9/openj9/issues/17085

Reply:
@keithc-ca I have fixed the other issues. Does the above info resolve your concerns?

Reply:
The biggest question for me is
> Why should we be here at all if dynamic agent loading is disabled?

Reply:
My understanding is that `DisableDynamicAgentLoading` disables loading agents at runtime, however agents can still be attached at JVM startup
see https://mail.openjdk.org/pipermail/serviceability-dev/2023-March/046868.html

I've also moved the `omrthread_monitor_enter`/`exit` code as mentioned before which I missed

Reply:
I think improving the comment on line 710 would solve my remaining concerns; yes, those flags were cleared earlier, but explaining why we want to set them now would help.

Reply:
I'm happy with the comment now; thank you.